### ISSUE TITLE
[wavedeform] static analysis issue (Trac #1194)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1188]
The static analysis found a potential error:
http://software.icecube.wisc.edu/static_analysis/00_LATEST/report-0637e0.html#EndPath

Looks like "Array subscript is undefined" when going down a particular branch.

Migrated from https://code.icecube.wisc.edu/ticket/1194
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:57", 
    "description": "The static analysis found a potential error:\nhttp://software.icecube.wisc.edu/static_analysis/00_LATEST/report-0637e0.html#EndPath\n\nLooks like \"Array subscript is undefined\" when going down a particular branch.\n", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067117911749", 
    "component": "combo reconstruction", 
    "summary": "[wavedeform] static analysis issue", 
    "priority": "critical", 
    "keywords": "", 
    "time": "2015-08-19T17:05:34", 
    "milestone": "", 
    "owner": "jbraun", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150820T13:56:44: jbraun commented:

> The static analysis picked up that passing argument npos < 0 would cause a segfault in nnls.c.  Use unsigned types instead, since, fundamentally, we're dealing with counts, which do not have sign.

**Comment by icecube-trac** [comment]
Trac update at 20150820T14:03:41: 
* jbraun changed resolution from "" to "fixed"
* jbraun changed status from "new" to "closed"
* jbraun commented:

> In changeset 136614:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="136614"
> This closes #1188, but I'm sure the static analysis will find other ways to break this code.  Make those separate issues.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:57: 
* olivas commented:

> Milestone IceRec V5 deleted


* olivas changed milestone from "IceRec V5" to ""
